### ISSUE TITLE
Allow <amp-accordion> to use non-<h*> elements

### ISSUE THREAD

**Issue by nainar** [ampproject/amphtml#25661]
"The first element in an <amp-accordion> - the one you need to click on to expand it - needs to be <h*>.
Is there a reason for this?
It makes it difficult for some companies to meet their SEO needs.

cc @morsssss

**Comment by morsssss** [comment]
I think that something ate the component name in that post... this happens sometimes!

The first sentence should be, "The first element in an `<amp-accordion>` - the one you need to click on to expand it - needs to be `<h*>`."

**Comment by morsssss** [comment]
cc @SimonHogstromRonbol

**Comment by sparhami** [comment]
The PR that adds the validation restricting this is #2634, which comes from this commit: https://github.com/ampproject/amphtml/pull/2634/commits/b14057c0bac1b8e0e954f7793913cfb6e82db727, though there is no context on why the restriction was added.

**Comment by morsssss** [comment]
I've just been guessing that the idea was to make the markup more semantic. But that there wouldn't be a technical reason for this. Of course, someone else could easily prove me wrong 😸

**Comment by morsssss** [comment]
Just pinging this one up again. I'm guessing it wouldn't be hard to implement. (famous last words etc)

**Comment by caroqliu** [comment]
Dedup with #21046

**Comment by morsssss** [comment]
Coolness!
